File: lab4/zad1/Figury.java
import Figura.*;
import java.util.Arrays;

public class Figury {
	public static void main(String[] args) {
		boolean[] is_f = new boolean[args[0].length()];
		Arrays.fill(is_f, true);
		Figura[] f = new Figura[args[0].length()];
		int arg = 1;
		for (int i = 0; i < args[0].length(); ++i) {
			if (arg == args.length) {
				System.out.println("brakuje argumentow");
				return;
			}
			char c = args[0].charAt(i);
			switch (c) {
				case 'o':
				{
					float a = 0;
					try {
						a = Float.parseFloat(args[arg]);
					} catch (IllegalArgumentException e) {
						System.out.println("o - " + args[arg] + " - dana figura nie istnieje");
						is_f[i] = false;
						++arg;
						break;
					}
					if (a < 0.0f) {
						System.out.println("o - " + a + " - dana figura nie istnieje");
						is_f[i] = false;
						++arg;
						break;
					}
					f[i] = new Kolo(a);
					++arg;
					break;
				}
				case 'c':
				{
					if (arg + 4 >= args.length) {
						System.out.println("brakuje argumentow");
						return;
					}
					float[] a = new float[5];
					boolean parsed = true;
					for (int j = 0; j < 5; ++j) {
						try {
							a[j] = Float.parseFloat(args[arg + j]);
						} catch (IllegalArgumentException e) {
							System.out.println("o - " + args[arg] + " " + args[arg + 1] + " " + args[arg + 2] + " " + args[arg + 3] + " " + args[arg + 4] + " - dana figura nie istnieje");
							parsed = false;
							is_f[i] = false;
							break;
						}
					}
					if (parsed == false) {
						break;
					}
					if (a[0] < 0.0f || a[1] < 0.0f || a[2] < 0.0f || a[3] < 0.0f || a[4] < 0.0f || a[4] > 90.0f) {
						System.out.println("c - " + args[arg] + " " + args[arg + 1] + " " + args[arg + 2] + " " + args[arg + 3] + " " + args[arg + 4] +" - dana figura nie istnieje");
						is_f[i] = false;
						arg += 5;
						break;
					}
					if (a[4] == 90 && a[0] == a[1] && a[1] == a[2] && a[2] == a[3]) {
						f[i] = new Kwadrat(a[0], a[1], a[2], a[3], a[4]);
					} else if (a[4] == 90 && (a[0] == a[1] && a[2] == a[3])) {
						f[i] = new Prostokat(a[0], a[1], a[2], a[3], a[4]);
					} else if (a[4] == 90 && (a[0] == a[2] && a[1] == a[3])) {
						f[i] = new Prostokat(a[0], a[2], a[1], a[3], a[4]);
					} else if (a[4] == 90 && (a[0] == a[3] && a[1] == a[2])) {
						f[i] = new Prostokat(a[0], a[3], a[1], a[2], a[4]);
					} else if (a[0] == a[1] && a[1] == a[2] && a[2] == a[3]) {
						f[i] = new Romb(a[0], a[1], a[2], a[3], a[4]);
					} else {
						System.out.println(args[arg] + " " + args[arg + 1] + " " + args[arg + 2] + " " + args[arg + 3] + " " + args[arg + 4]  + " - dany czworokat nie istnieje");
						is_f[i] = false;
					}
					arg += 5;
					break;
				}
				case 'p':
				{
					float a = 0;
					try {
						a = Float.parseFloat(args[arg]);
					} catch (IllegalArgumentException e) {
						System.out.println("p - " + args[arg] + " - dana figura nie istnieje");
						is_f[i] = false;
						++arg;
						break;
					}
					if (a < 0.0f) {	
						System.out.println("p - " + a + " - dana figura nie istnieje");
						is_f[i] = false;
						++arg;
						break;
					}
					f[i] = new Pieciokat(a);
					++arg;
					break;
				}
				case 's':
				{
					float a = 0;
					try {
						a = Float.parseFloat(args[arg]);
					} catch (IllegalArgumentException e) {
						System.out.println("s - " + args[arg] + " - dana figura nie istnieje");
						is_f[i] = false;
						++arg;
						break;
					}
					if (a < 0.0f) {	
						System.out.println("s - " + a + " - dana figura nie istnieje");
						is_f[i] = false;
						++arg;
						break;
					}
					f[i] = new Szesciokat(a);
					++arg;
					break;
				}
				default:
					System.out.println(c + " - figura nie istnieje");
					is_f[i] = false;
			}
			if (is_f[i] == true) {
				System.out.println("Pole = " + f[i].calcArea() + ", Obwod = " + f[i].calcCircumference());
			}
		}
	}
}

File: lab4/zad2/figury.cpp
#include "figura.hpp"
#include <iostream>
#include <algorithm>
//vector
#include <cstdlib>
#include <cstring>

int main(int argc, char* argv[]) {
	int l = strlen(argv[1]);
	//bool is_f[l];
	//std::fill(is_f, is_f + l, true);
	//Figura f[l];
	int arg = 2;
	for (int i = 0; i < l; ++i) {
		if (arg == argc) {
			std::cout << "brakuje argumentow\n";
			return 0;
		}
		char c = argv[1][i];
		switch (c) {
			case 'o':
			{
				char* c;
				float a = std::strtof(argv[arg], &c);
				if (a <= 0) {
					std::cout << "o - " << argv[arg] << " - dana figura nie istnieje\n";
					//is_f[i] = false;
					++arg;
					break;
				}
				Kolo k(a);
				//f[i] = k;
				std::cout << "Pole = " << k.calcArea() << ", Obwod = " << k.calcCircumference() << "\n";
				++arg;
				break;
			}
			case 'c':
			{
				if (arg + 4 >= argc) {
					std::cout << "brakuje argumentow\n";
					return 0;
				}
				float a[5];
				for (int j = 0; j < 5; ++j) {
					char* c;
					a[j] = std::strtof(argv[arg + j], &c);
				}
				if (a[0] <= 0 || a[1] <= 0 || a[2] <= 0 || a[3] <= 0 || a[4] <= 0 || a[4] > 90) {
					std::cout << "c - " << argv[arg] << " " << argv[arg + 1] << " " << argv[arg + 2] << " " << argv[arg + 3] << " " << argv[arg + 4] << " - dana figura nie istnieje\n";
					//is_f[i] = false;
					arg += 5;
					break;
				}
				if (a[4] == 90 && a[0] == a[1] && a[1] == a[2] && a[2] == a[3]) {
					Kwadrat k(a[0], a[1], a[2], a[3], a[4]);
					std::cout << "Pole = " << k.calcArea() << ", Obwod = " << k.calcCircumference() << "\n";
				} else if (a[4] == 90 && a[0] == a[1] && a[2] == a[3]) {
					Prostokat p(a[0], a[1], a[2], a[3], a[4]);
					std::cout << "Pole = " << p.calcArea() << ", Obwod = " << p.calcCircumference() << "\n";
				} else if (a[4] == 90 && a[0] == a[2] && a[1] == a[3]) {
					Prostokat p(a[0], a[2], a[1], a[3], a[4]);
					std::cout << "Pole = " << p.calcArea() << ", Obwod = " << p.calcCircumference() << "\n";
				} else if (a[4] == 90 && a[0] == a[3] && a[1] == a[2]) {
					Prostokat p(a[0], a[3], a[1], a[2], a[4]);
					std::cout << "Pole = " << p.calcArea() << ", Obwod = " << p.calcCircumference() << "\n";
				} else if (a[0] == a[1] && a[1] == a[2] && a[2] == a[3]) {
					Romb r(a[0], a[1], a[2], a[3], a[4]);
					std::cout << "Pole = " << r.calcArea() << ", Obwod = " << r.calcCircumference() << "\n";
				} else {
					std::cout << a[0] << " " << a[1] << " " << a[2] << " " << a[3] << " " << a[4] << " - dany czworokat nie istnieje\n";
					//is_f[i] = false;
				}
				arg += 5;
				break;
			}
			case 'p':
			{
				char* c;
				float a = std::strtof(argv[arg], &c);
				if (a <= 0) {
					std::cout << "p - " << argv[arg] << "- dana figura nie istnieje\n";
					//is_f[i] = false;
					++arg;
					break;
				}
				Pieciokat p(a);
				std::cout << "Pole = " << p.calcArea() << ", Obwod = " << p.calcCircumference() << "\n";
				++arg;
				break;
			}
			case 's':
			{
				char* c;
				float a = std::strtof(argv[arg], &c);
				if (a <= 0) {
					std::cout << "s - " << argv[arg] << " - dana figura nie istnieje\n";
					//is_f[i] = false;
					++arg;
					break;
				}
				Szesciokat s(a);
				std::cout << "Pole = " << s.calcArea() << ", Obwod = " << s.calcCircumference() << "\n";
				++arg;
				break;
			}
			default:
				std::cout << c << " - figura nie istnieje\n";
				//is_f[i] = false;
		}
	}
}
